### PR TITLE
Add Bot identity To TurnState for Getting Token

### LINKED ISSUE #3444: Linked accounts is no longer working
#### What project is affected?
Linked accounts in master
#### What language is this in?
C#
#### What happens?
It is no longer working
#### What are the steps to reproduce this issue?
* Setup according to document https://microsoft.github.io/botframework-solutions/solution-accelerators/samples/linked-accounts/
* When opening, 'An IIdentity is required in TurnState for this operation.' error https://github.com/microsoft/botbuilder-dotnet/blob/03a51e1e09d36cd8c4e8f1dcde6263e5d62e806e/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L1585
* After setting IIdentity, it will be 'bad request' because 'user id is incorrect'. It seems the server has high security for generating sign in url now.

### PR DESCRIPTION
**Pull request by sidmore**
### Purpose
*What is the context of this pull request? Why is it being done?*
Add BotIdentity To TurnState

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: samples/csharp/clients/linkedaccounts.web/Controllers/LinkedAccountRepository.cs [hunk 1]
```diff
@@ -37,6 +40,11 @@ public async Task<TokenStatus[]> GetTokenStatusAsync(string userId, ICredentialP
             {
                 var connectorClient = new ConnectorClient(new Uri(TokenServiceUrl), botAppId, botAppPassword);
                 context.TurnState.Add<IConnectorClient>(connectorClient);
+                context.TurnState.Add<IIdentity>("BotIdentity", new ClaimsIdentity(new List<Claim>
+                {
+                    new Claim(AuthenticationConstants.AudienceClaim, botAppId),
```
Comment (siddimore) [inline samples/csharp/clients/linkedaccounts.web/Controllers/LinkedAccountRepository.cs:45]: Do we need AudienceClaim here?

Comment (sidmore) [inline samples/csharp/clients/linkedaccounts.web/Controllers/LinkedAccountRepository.cs:46]: Do we need AudienceClaim here?

**Review by pavolum — APPROVED** [review]

File: samples/csharp/clients/linkedaccounts.web/Controllers/LinkedAccountRepository.cs (hunk 1 above)
Comment (lauren-mills) [reply samples/csharp/clients/linkedaccounts.web/Controllers/LinkedAccountRepository.cs:46]: This code in the BotFrameworkAdapter looks like only one or the other is required: https://github.com/microsoft/botbuilder-dotnet/blob/2ea5614ef8c01ad122454f182cd77b99e39d2931/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L1469

**Comment by xieofxie** [comment]
@sidmore I tested and it is still not working. Do I miss something?

I updated several parts according to https://blog.botframework.com/2018/09/25/enhanced-direct-line-authentication-features/ & https://github.com/microsoft/botbuilder-dotnet/blob/03a51e1e09d36cd8c4e8f1dcde6263e5d62e806e/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L1591 in here https://github.com/microsoft/botframework-solutions/tree/hualxie/issue_3444

The home page shows up but when I click Link, it will report
image
Thanks

**Comment by sidmore** [comment]
@xieofxie i used contoso identity and created a AAD app in that tenant to be able to use linked apps. Can you try that please?

**Review by taicchoumsft — APPROVED** [review]
:shipit:

**Review by darrenj — APPROVED** [review]

**Pull request merged**